### PR TITLE
Add `nvc++` as a C++ compiler

### PR DESCRIPTION
**Pull request by trxcllnt**
<details><summary>Adds nvc++ as a C++ compiler:</summary><a target="_blank" rel="noopener noreferrer nofollow" href="https://user-images.githubusercontent.com/178183/198147097-4365e480-2eba-4aa6-9108-7d18c2273b24.png">image</a></details>

Related infra PR: https://github.com/compiler-explorer/infra/pull/855

### CODE REVIEW

**Comment by jrhemstad** [comment]
@mattgodbolt friendly ping :)

**Comment by mattgodbolt** [comment]
@jrhemstad was waiting on @trxcllnt looking at the installers, but... I took the time to fix the installers. So...will be able to do this shortly.

**Comment by mattgodbolt** [comment]
Per Discord, the aarch64 ones aren't installing currently.

File: etc/config/c++.amazon.properties [hunk 1]
```diff
@@ -1,4 +1,4 @@
-compilers=&gcc86:&icc:&icx:&clang:&clangx86trunk:&clang-rocm:&rvclang:&wasmclang:&cl:&cross:&ellcc:&zapcc:&djggp:www.godbolt.ms@443:&armclang32:&armclang64:&zigcxx:&cxx6502
+compilers=&gcc86:&icc:&icx:&clang:&clangx86trunk:&clang-rocm:&rvclang:&wasmclang:&cl:&cross:&ellcc:&zapcc:&djggp:www.godbolt.ms@443:&armclang32:&armclang64:&zigcxx:&cxx6502:&nvcxx_x86_cxx:&nvcxx_arm_cxx
```
Comment (mattgodbolt) [inline etc/config/c++.amazon.properties:1]: The ARM compilers are failing to install right now; but I'll leave them in here for now as once they're working they'll appear after the next release.

**Review by mattgodbolt — APPROVED** [review]
Thanks!

**Pull request merged**